Allow users to add uploads to collections

SuaYoo opened the pull request:
Resolves https://github.com/webrecorder/browsertrix-cloud/issues/936

<!-- Fixes #issue_number -->

### Changes

Initial implementation of collection upload picker for new and existing collections.

### Screenshots

| Page | Image/video |
| ---- | ----------- |
| New Collection - Select Uploads | **Empty state**:<br/><img width="920" alt="Screen Shot 2023-07-08 at 9 11 43 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/c6292910-4d56-4fe5-9436-fdd02eafd49b"> |
| Edit Collection - Select Uploads | <img width="1077" alt="Screen Shot 2023-07-08 at 10 02 32 PM" src="https://github.com/webrecorder/browsertrix-cloud/assets/4672952/30409429-6b40-4f41-b6d5-0bab607a1970"> |



### Follow-ups

This is a quick-and-dirty implementation and isn't the most ideal UI, maybe drag-over, or integration with "Select Crawls" for the next release? Once upload UI is added to browsertrix cloud, we should link to it in the empty state and/or allow direct upload into a collection.

SuaYoo committed:
add section

SuaYoo committed:
handle empty state

SuaYoo committed:
update type

SuaYoo committed:
add selector

SuaYoo commented:
@ikreymer it looks like we need to add the `collectionId` filter to [GET /api/orgs/{oid}/uploads](https://dev.browsertrix.cloud/api/docs#/uploads/list_uploads_api_orgs__oid__uploads_get), like `GET /crawls`, in order to list the uploads in a collection.

ikreymer commented:
> @ikreymer it looks like we need to add the `collectionId` filter to [GET /api/orgs/{oid}/uploads](https://dev.browsertrix.cloud/api/docs#/uploads/list_uploads_api_orgs__oid__uploads_get), like `GET /crawls`, in order to list the uploads in a collection.

Done via #969

Almost working now, though seeing occasional empty left panel on initial load after coll:
<img width="1167" alt="Screen Shot 2023-07-09 at 10 54 11 AM" src="https://github.com/webrecorder/browsertrix-cloud/assets/1015759/ee562dec-1a9d-4b33-9698-8bcc22a37feb">

If easier, could just have only right panel and have it be checked.

ikreymer committed:
combine fetchCollectionCrawls() and fetchCollectionUploads() into single fetchCollectionCrawlsAndUploads()
which uses /all-crawls endpoint to get single query for all objects in collection (depends on #970)
avoids reassigning same value with different filters

ikreymer commented:
> Almost working now, though seeing occasional empty left panel on initial load after coll: <img alt="Screen Shot 2023-07-09 at 10 54 11 AM" width="1167" src="https://user-images.githubusercontent.com/1015759/252160942-ee562dec-1a9d-4b33-9698-8bcc22a37feb.png">
> 

Also added `?collectionId=` filter to `/all-crawls`, switched to using single all-crawls query instead of two queries, think that fixes this issue, seemed to be caused by double-init of filtered crawl list..
